diff --git a/aom_ports/aom_once.h b/aom_ports/aom_once.h
index 680120f..4cc2be7 100644
--- a/aom_ports/aom_once.h
+++ b/aom_ports/aom_once.h
@@ -38,7 +38,14 @@
  * by the call to foo().
  */
 
-#if CONFIG_MULTITHREAD && defined(_WIN32)
+#if CONFIG_MULTITHREAD && HAVE_PTHREAD_H
+#include <pthread.h>
+static void aom_once(void (*func)(void)) {
+  static pthread_once_t lock = PTHREAD_ONCE_INIT;
+  pthread_once(&lock, func);
+}
+
+#elif CONFIG_MULTITHREAD && defined(_WIN32)
 #undef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
@@ -60,13 +67,6 @@ static void aom_once(void (*func)(void)) {
   InitOnceComplete(&aom_init_once, 0, NULL);
 }
 
-#elif CONFIG_MULTITHREAD && HAVE_PTHREAD_H
-#include <pthread.h>
-static void aom_once(void (*func)(void)) {
-  static pthread_once_t lock = PTHREAD_ONCE_INIT;
-  pthread_once(&lock, func);
-}
-
 #else
 /* Default version that performs no synchronization. */
 
diff --git a/cmake/aom_configure.cmake b/cmake/aom_configure.cmake
index ac3e132..01fff98 100644
--- a/cmake/aom_configure.cmake
+++ b/cmake/aom_configure.cmake
@@ -240,7 +240,7 @@ endif()
 if(AOM_TARGET_SYSTEM STREQUAL "Windows")
   # The default _WIN32_WINNT value in MinGW is 0x0502 (Windows XP with SP2). Set
   # it to 0x0601 (Windows 7).
-  add_compiler_flag_if_supported("-D_WIN32_WINNT=0x0601")
+  #add_compiler_flag_if_supported("-D_WIN32_WINNT=0x0601")
   # Quiet warnings related to fopen, printf, etc.
   add_compiler_flag_if_supported("-D_CRT_SECURE_NO_WARNINGS")
 endif()
